compliance_checker/util.py

```diff
@@ -1,11 +1,27 @@
+"""
+General purpose utility functions to aid in compliance checking tasks
+"""
 import isodate
-import re
-"""General purpose utility functions to aid in compliance checking tasks"""
+import arrow
+
 
 def datetime_is_iso(dt):
     """Attempts to parse a date formatted in ISO 8601 format"""
     try:
         isodate.parse_datetime(dt)
         return True, []
     except isodate.ISO8601Error:
-        return False, ['Datetime provided is not in a valid ISO 8601 format']
+        return False, ['Datetime provided is not in a valid ISO 8602 format']
+
+
+def dateparse(date_str):
+    '''
+    Returns a datetime string parsed from an ISO-8601 input
+
+    :param str date_str: An ISO-8601 string
+    '''
+    if isinstance(date_str, basestring):
```

Comment:
Try to import `basestring` and catch the `ImportError` to assign `basestring` to `str` ( no acces to computer atm).

IMO this is the best date parser: https://github.com/ioos/pyoos/blob/master/pyoos/utils/asatime.py

I've never heard of isodate package!